api/categorisation.go

```diff
@@ -5,32 +5,34 @@ import (
 
 	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular"
 	"github.com/ONSdigital/dp-api-clients-go/v2/cantabularmetadata"
+	"github.com/ONSdigital/dp-cantabular-filter-flex-api/model"
 	"github.com/pkg/errors"
 )
 
-func (api *API) CheckDefaultCategorisation(dimName string, datasetName string) (string, string, error) {
+// RetrieveDefaultCategorisation takes dimension, returns categorisations, and checks if any are default.
+// if so, it returns the relevant information. If there are no default categorisations, it returns empty string
+// for default categorisation, and the original dimension name and label to persist instead.
+func (api *API) RetrieveDefaultCategorisation(dimension *model.Dimension, datasetName string) (string, string, string, error) {
```

Comment:
Preference for return will be either to introduce a struct or named return variables. Function signature is not obvious enough to explain what these 3 returned strings are

Reply:
I agree

Reply:
I think the named variables add too many extra lines of code, and a struct feels overkill for just one function, @j-s-rawat  can I just add a comment that explains the return signature?

Reply:
To be honest this seems like a bit of a code smell that the function is trying to do too much. It may be better to split it up into two (or more).

Reply:
given the need for the changes soon, this will be dealt with in a future refactor